runtime/doc/lua-plugin.txt

```diff
@@ -250,6 +259,32 @@ Validations could include:
   This can be tricky to implement, and may be better suited for a |health|
   check, to reduce overhead.
 
+==============================================================================
+UI                                                           *lua-plugin-ui*
+
+Some plugins have their own "UI" which they present in a buffer that the
+plugin "owns". In that buffer typically you will want to provide custom
+actions.
+
+Besides creating |<Plug>| mappings, you may want to consider providing actions
+by defining an in-process LSP "server". Offering actions as "code-actions"
```

Comment:
I mentioned this before but I find the quoting confusing. They make me think that the plugin doesn't really own the buffer or that the in-process LSP server isn't a real server.